homeassistant/components/nzbget/sensor.py

```diff
@@ -34,90 +36,80 @@
+        self,
+        coordinator: NZBGetDataUpdateCoordinator,
+        entry_id: str,
+        entry_name: str,
+        sensor_type: str,
+        sensor_name: str,
+        unit_of_measurement: Optional[str] = None,
     ):
         """Initialize a new NZBGet sensor."""
-        self._name = f"{client_name} {sensor_name}"
-        self.type = sensor_type
-        self.client_name = client_name
-        self.nzbget_data = nzbget_data
-        self._state = None
+        self._sensor_type = sensor_type
+        self._unique_id = f"{entry_id}_{sensor_type}"
         self._unit_of_measurement = unit_of_measurement
 
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return self._name
+        super().__init__(
+            coordinator=coordinator,
+            entry_id=entry_id,
+            name=f"{entry_name} {sensor_name}",
+        )
 
     @property
-    def state(self):
-        """Return the state of the sensor."""
-        return self._state
+    def unique_id(self) -> str:
+        """Return the unique ID of the sensor."""
+        return self._unique_id
 
     @property
-    def unit_of_measurement(self):
-        """Return the unit of measurement of this entity, if any."""
+    def unit_of_measurement(self) -> str:
+        """Return the unit that the state of sensor is expressed in."""
         return self._unit_of_measurement
 
     @property
-    def available(self):
-        """Return whether the sensor is available."""
-        return self.nzbget_data.available
-
-    async def async_added_to_hass(self):
-        """Handle entity which will be added."""
-        self.async_on_remove(
-            async_dispatcher_connect(
-                self.hass, DATA_UPDATED, self._schedule_immediate_update
-            )
-        )
-
-    @callback
-    def _schedule_immediate_update(self):
-        self.async_schedule_update_ha_state(True)
-
-    def update(self):
-        """Update state of sensor."""
-
-        if self.nzbget_data.status is None:
-            _LOGGER.debug(
-                "Update of %s requested, but no status is available", self._name
-            )
-            return
+    def state(self):
+        """Return the state of the sensor."""
+        value = self.coordinator.data.status.get(self._sensor_type)
 
-        value = self.nzbget_data.status.get(self.type)
         if value is None:
-            _LOGGER.warning("Unable to locate value for %s", self.type)
-            return
+            _LOGGER.warning("Unable to locate value for %s", self._sensor_type)
+            return None
 
-        if "DownloadRate" in self.type and value > 0:
+        if "DownloadRate" in self._sensor_type and value > 0:
             # Convert download rate from Bytes/s to MBytes/s
-            self._state = round(value / 2 ** 20, 2)
-        elif "UpTimeSec" in self.type and value > 0:
+            return round(value / 2 ** 20, 2)
+
+        if "UpTimeSec" in self._sensor_type and value > 0:
```

Comment:
The uptime sensor should preferably be a timestamp sensor instead. We don't want relative times in the state machine.